### ISSUE TITLE
[BUG] Error: Missing required argument #1 on Linux Mint, 5.3.0-59-generic Kernal

### ISSUE THREAD

**Issue by msglm** [npm/cli#1456]
# What / Why

> When using npm install, the installer fails with the following error

```
npm ERR! typeerror Error: Missing required argument #1
npm ERR! typeerror     at andLogAndFinish (/usr/share/npm/lib/fetch-package-metadata.js:31:3)
npm ERR! typeerror     at fetchPackageMetadata (/usr/share/npm/lib/fetch-package-metadata.js:51:22)
npm ERR! typeerror     at resolveWithNewModule (/usr/share/npm/lib/install/deps.js:456:12)
npm ERR! typeerror     at /usr/share/npm/lib/install/deps.js:457:7
npm ERR! typeerror     at /usr/share/npm/node_modules/iferr/index.js:13:50
npm ERR! typeerror     at /usr/share/npm/lib/fetch-package-metadata.js:37:12
npm ERR! typeerror     at addRequestedAndFinish (/usr/share/npm/lib/fetch-package-metadata.js:82:5)
npm ERR! typeerror     at returnAndAddMetadata (/usr/share/npm/lib/fetch-package-metadata.js:117:7)
npm ERR! typeerror     at pickVersionFromRegistryDocument (/usr/share/npm/lib/fetch-package-metadata.js:134:20)
npm ERR! typeerror     at /usr/share/npm/node_modules/iferr/index.js:13:50
npm ERR! typeerror This is an error with npm itself. Please report this error at:
npm ERR! typeerror     <http://github.com/npm/npm/issues>
```

## When
* Anytime upon install

## Where

*I do not understand the requested info, only worked with npm for building software.  

### Current Behavior
>type in npm install 
>crash with already posted error
* n/a

### Steps to Reproduce
>Run Linux Mint 19.3 Tricia x86_64 with 5.3.0-59-generic Kernal.
>Run npm install (https://github.com/Nicklason/tf2-automatic specifically on this, doubt this is relavent, but good to know if the problem is a false positive for being npm's fault)
>wait for it to crash

### Expected Behavior

* installing of modules, im not too savy with this and do not understand how it should work or what it should do.

[npm-debug.log](https://github.com/npm/cli/files/4818913/npm-debug.log)

**Comment by jonchurch** [comment]
Related https://github.com/npm/cli/issues/681

This issue seems to pop up on the specific version of npm you’re using `3.5.2`, which is a few years old now.

The quick fix is updating to the latest release in the 3.x line `npm i -g npm@3`.

But the proper fix is to update to the LTS version of Node.js and npm.

**Comment by msglm** [comment]
I updated and ran the command on the modules im trying to build, similar error as last time.
Did want to point out that the following appeared at the top in most error cases:
```
npm WARN deprecated request@2.88.2: request has been deprecated, see 
https://github.com/request/request/issues/3142
npm ERR! Linux 5.3.0-59-generic
```
Don't know if that is apart of the issue, but seems to be worth mentioning.  Updated debug log is below.

[npm-debug.log](https://github.com/npm/cli/files/4819105/npm-debug.log)
